pcsx2/Patch.cpp

```diff
@@ -254,9 +258,28 @@ namespace PatchFunc
 		}
 
 		iPatch.cpu = (patch_cpu_type)PatchTableExecute(pieces[1], std::string_view(), cpuCore);
-		iPatch.addr = StringUtil::FromChars<u32>(pieces[2], 16).value_or(0);
 		iPatch.type = (patch_data_type)PatchTableExecute(pieces[3], std::string_view(), dataType);
-		iPatch.data = StringUtil::FromChars<u64>(pieces[4], 16).value_or(0);
+
+
+		if (StringUtil::StartsWithNoCase(pieces[2], "0x"))
```

Comment:
I think ideally we should split this out into a common helper, something like `StringUtil::ParseBasePrefixString()` or something (that way it could handle `0x`, `0b`, and `0` prefixes for 16/1/8 bases respectively).